### PR TITLE
removed shell=True

### PR DESCRIPTION
**Pull request by mzainea**
Refactored audit/misc.py and hubblestack/misc.py to remove bash code that was being called from python and remove `shell=True` and `python_shell=True` statements.

### CODE REVIEW

File: hubblestack/files/hubblestack_nova/misc.py [hunk 1]
```diff
@@ -400,55 +410,40 @@ def check_path_integrity(reason=''):
     """
     Ensure that system PATH variable is not malformed.
     """
+    path_value = os.environ.get('PATH')
+    output = ''
+    for item in path_value.split(':'):
```
Comment (vladmonea) [inline hubblestack/files/hubblestack_nova/misc.py:415]: if `path_value` is `None`, which is the default value should the `'PATH'` key not be present in `os.environ`, this call will fail. Maybe ensure an appropriate default value is given?

File: hubblestack/files/hubblestack_nova/misc.py [hunk 2]
```diff
@@ -511,15 +506,17 @@ def check_core_dumps(reason=''):
...
+    # hard_core_dump_value = _execute_shell_command("grep -R -E \"hard +core\" /etc/security/limits.conf /etc/security/limits.d/ | awk '{print $4}'", python_shell=True).strip()
+    # hard_core_dump_value = hard_core_dump_value.split('\n') if hard_core_dump_value != "" else []
+    # if '0' in hard_core_dump_value:
+    #     return True
+    #
+    # if hard_core_dump_value is None or hard_core_dump_value == [] or hard_core_dump_value == "":
+    #     return "'hard core' not found in any file"
+    #
+    # return str(hard_core_dump_value)
+    return True
```
Comment (vladmonea) [inline hubblestack/files/hubblestack_nova/misc.py:519]: Why is this just being commented out and replaced with `return True`?

File: hubblestack/files/hubblestack_nova/misc.py [hunk 3]
```diff
@@ -546,58 +543,49 @@ def check_ssh_timeout_config(reason=''):
...
     """
     Ensure all users' home directories exist
     """
-
-    max_system_uid = int(max_system_uid)
-    users_uids_dirs = _execute_shell_command("cat /etc/passwd | awk -F: '{ print $1 \" \" $3 \" \" $6 \" \" $7}'", python_shell=True).strip()
-    users_uids_dirs = users_uids_dirs.split('\n') if users_uids_dirs else []
+    with open('/etc/passwd', 'r') as passwd:
+        lines = passwd.readlines()
+        users_uids_dirs = [' '.join([item.split(':')[0], item.split(':')[2], item.split(':')[5], item.split(':')[6].strip()]) for item in lines]
```
Comment (vladmonea) [inline hubblestack/files/hubblestack_nova/misc.py:588]: Maybe, for clarity:
```suggestion
        items = item.split(':')[0], item.split(':')[2], item.split(':')[5], item.split(':')[6].strip()
        users_uids_dirs = [' '.join(items) for item in lines]
```
You decide, this is just a suggestion.

**Review by vladmonea — CHANGES_REQUESTED** [review]

File: hubblestack/files/hubblestack_nova/misc.py [hunk 4]
```diff
@@ -361,26 +370,27 @@ def test_mount_attrs(mount_name, attribute, check_type='hard'):
...
-        return "neither ntpd nor timesyncd is running"
-    else:
-        return True
+    manager = Manager()
+    manager.load()
+    services = manager.Manager.ListUnitFiles()
+    success = any([item for item in services if b'systemd-timesyncd' in item[0]
+                                                or b'ntpd' in item[0]
+                                                and b'enabled' in item[1]])
+    return success or 'neither ntpd nor timesyncd is running'
```
[the remark this answers is not in the document]
Comment (mzainea) [reply hubblestack/files/hubblestack_nova/misc.py:393]: yeah it's a case of the truthies, we'll deffinetly have to tug on this string and see what it unravels

File: hubblestack/files/hubblestack_nova/misc.py (hunk 2 above)
Comment (mzainea) [reply hubblestack/files/hubblestack_nova/misc.py:519]: the check the code runs is successful even for documentation lines which are commented out, a little more research is needed to fix it but in the interim I commented it out so it stops returning incorrect results

File: hubblestack/files/hubblestack_nova/misc.py [hunk 5]
```diff
@@ -326,25 +326,34 @@ def sticky_bit_on_world_writable_dirs(reason=''):
...
 
 def root_is_only_uid_0_account(reason=''):
     """
     Ensure root is the only UID 0 account
     """
-    result = _execute_shell_command('cat /etc/passwd | awk -F: \'($3 == 0) { print $1 }\'', python_shell=True)
-    return True if result.strip() == 'root' else result
+    uid0_accounts = []
+    with open('/etc/passwd', 'r') as passwd:
+        lines = passwd.readlines()
```
[the remark this answers is not in the document]
Comment (mzainea) [reply hubblestack/files/hubblestack_nova/misc.py]: fixed this

**Pull request closed without merging**